File: services/trip-service/internal/application/dto/calendar_sync.go
package dto

import "time"

type TripCalendarSyncStatus struct {
	Provider                 string     `json:"provider"`
	Connected                bool       `json:"connected"`
	ProviderAccountEmail     *string    `json:"providerAccountEmail,omitempty"`
	Synced                   bool       `json:"synced"`
	LastSyncedAt             *time.Time `json:"lastSyncedAt,omitempty"`
	SyncedItineraryRevision  int        `json:"syncedItineraryRevision,omitempty"`
	CurrentItineraryRevision int        `json:"currentItineraryRevision"`
	OutOfDate                bool       `json:"outOfDate"`
	EventCount               int        `json:"eventCount"`
}

type TripCalendarSyncResult struct {
	Provider          string     `json:"provider"`
	Status            string     `json:"status"`
	Created           int        `json:"created"`
	Updated           int        `json:"updated"`
	Deleted           int        `json:"deleted"`
	Failed            int        `json:"failed"`
	Skipped           int        `json:"skipped"`
	ItineraryRevision int        `json:"itineraryRevision"`
	LastSyncedAt      *time.Time `json:"lastSyncedAt,omitempty"`
}

type TripCalendarDeleteResult struct {
	Provider string `json:"provider"`
	Deleted  int    `json:"deleted"`
	Failed   int    `json:"failed"`
}

File: apps/web/src/entities/calendar-sync/model/calendar-sync.ts
export type CalendarConnectionStatus = {
  connected: boolean;
  provider: "google";
  providerAccountEmail?: string | null;
  connectedAt?: string | null;
  scopes?: string | null;
};

export type TripCalendarSyncStatus = {
  provider: "google";
  connected: boolean;
  providerAccountEmail?: string | null;
  synced: boolean;
  lastSyncedAt?: string | null;
  syncedItineraryRevision?: number;
  currentItineraryRevision: number;
  outOfDate: boolean;
  eventCount: number;
};

export type TripCalendarSyncResult = {
  provider: "google";
  status: "synced" | "no_timed_items";
  created: number;
  updated: number;
  deleted: number;
  failed: number;
  skipped?: number;
  itineraryRevision: number;
  lastSyncedAt?: string | null;
};

export type TripCalendarDeleteResult = {
  provider: "google";
  deleted: number;
  failed: number;
};
